packages/react-aria-components/src/Collection.tsx

```diff
@@ -642,17 +649,22 @@ export class Document<T, C extends BaseCollection<T> = BaseCollection<T>> extend
 
 export interface CollectionProps<T> extends Omit<CollectionBase<T>, 'children'> {
   /** The contents of the collection. */
-  children?: ReactNode | ((item: T) => ReactNode)
+  children?: ReactNode | ((item: T) => ReactNode),
+  /** Values that should invalidate the item cache when using dynamic collections. */
```

Comment:
want to add anything about how it should work like the deps array for useMemo? aka, order of items shouldn't change, items shouldn't be added or removed, etc?

Reply:
Maybe not here but we should probably add something in the docs somewhere at some point

Reply:
how certain are we about supporting this `dependency` prop in other RAC Collection components since CollectionProps is used in those components too? On one hand, it seems like it doesn't hurt to have it available for ListBox/GridList/etc but if our only use case for it is for Table then do we want to only explicitly expose it on TableHeader/TableBody/Row etc?

Reply:
Yeah I think it's useful as an escape hatch in general. The same issue can also be seen in any case where the render function uses any inputs other than the item object itself, e.g. other state. A similar case I found was with event handlers that indirectly use other state.